### PR TITLE
submodule: bump third_party/openthread/repo from `2796615` to `8467a3d`

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Review by bukepo — APPROVED** [review]
LGTM:+1:

File: src/agent/border_agent.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@
 #include <string.h>
 #include <unistd.h>
 
-#include <openthread/thread_ftd.h>
 #include <openthread/border_agent.h>
+#include <openthread/thread_ftd.h>
```
Comment (bukepo) [inline src/agent/border_agent.cpp:46]: does t his need to be sorted?

Comment (jwhui) [reply src/agent/border_agent.cpp:46]: It fails pretty check:
https://github.com/openthread/ot-br-posix/runs/2397393075?check_suite_focus=true

Comment (bukepo) [reply src/agent/border_agent.cpp:46]: weird, I thought `t`hread should be after `p`platform.

**Comment by jwhui** [comment]
@bukepo @wgtdkp , meshcop test is failing - can one of you take a look?

https://github.com/openthread/ot-br-posix/pull/806/checks?check_run_id=2397508535

**Comment by bukepo** [comment]
> @bukepo @wgtdkp , meshcop test is failing - can one of you take a look?
> 
> https://github.com/openthread/ot-br-posix/pull/806/checks?check_run_id=2397508535

I think it caused by parsing the port. I'll add a commit.

**Comment by bukepo** [comment]
submitted a fix for grep.

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

**Pull request merged**